cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -27,19 +32,139 @@
  */
 public class BasicWorkflowToken implements WorkflowToken {
   private Map<String, Map<String, Long>> mapReduceCounters;
+  private Map<String, List<NodeValueEntry>> tokenValueMap = Maps.newHashMap();
```

Comment:
Another question not related to the field signature. I think it's better for the actual instance of the List be immutable.